test/inheritAttributes.test.js

```diff
@@ -2238,3 +2250,118 @@ QUnit.test('keySystem info for representation - uppercase UUIDs', function(asser
+          <Event presentationTime="1900" duration="0" id="6" messageData="foo_bar"/>
+        </EventStream>
+      </Period>
+    </MPD>`);
+  const expected = [
+    {
+      end: 2.1,
+      id: '0',
+      messageData: 'foo',
+      schemeIdUri: 'urn:google:dai:2018',
+      start: 2.1,
+      value: undefined
+    },
+    {
+      end: 2.9,
+      id: '5',
+      messageData: 'bar',
+      schemeIdUri: 'urn:google:dai:2018',
+      start: 2.9,
+      value: undefined
+    },
+    {
+      end: 3.9,
+      id: '6',
+      messageData: 'foo_bar',
+      schemeIdUri: 'urn:google:dai:2018',
+      start: 3.9,
+      value: undefined
+    }
+  ];
+
+  const firstPeriod = { node: findChildren(mpd, 'Period')[0], attributes: { start: 2 } };
+  const eventStreams = toEventStream(firstPeriod);
+
+  assert.deepEqual(eventStreams, expected, 'toEventStream returns the expected object');
+});
+
+QUnit.test('cannot get EventStream data from toEventStream with no schemeIdUri', function(assert) {
+  const mpd = stringToMpdXml(`
+    <MPD mediaPresentationDuration="PT30S" xmlns:cenc="urn:mpeg:cenc:2013">
+      <Period id="dai_pod-0001065804-ad-1" start="PT17738H17M14.156S" duration="PT9.977S">
+        <BaseURL>https://www.example.com/base/</BaseURL>
+        <SegmentTemplate media="$RepresentationID$/$Number$.mp4" initialization="$RepresentationID$/init.mp4"/>
+        <EventStream timescale="1000">
+          <Event presentationTime="100" duration="0" id="0" messageData="foo"/>
+          <Event presentationTime="900" duration="0" id="5" messageData="bar"/>
+          <Event presentationTime="1900" duration="0" id="6" messageData="foo_bar"/>
+        </EventStream>
+      </Period>
+    </MPD>`);
+
+  const firstPeriod = { node: findChildren(mpd, 'Period')[0], attributes: { start: 2} };
+  const eventStreams = toEventStream(firstPeriod);
+
+  assert.deepEqual(eventStreams, [undefined], 'toEventStream returns the expected object');
+});
+
+QUnit.test('gets eventStream from inheritAttributes', function(assert) {
+  const mpd = stringToMpdXml(`
+    <MPD mediaPresentationDuration="PT30S" xmlns:cenc="urn:mpeg:cenc:2013">
+      <Period id="dai_pod-0001065804-ad-1" start="PT0H0M14.9S" duration="PT9.977S">
+        <BaseURL>https://www.example.com/base/</BaseURL>
+        <SegmentTemplate media="$RepresentationID$/$Number$.mp4" initialization="$RepresentationID$/init.mp4"/>
+        <EventStream schemeIdUri="urn:google:dai:2018" timescale="1000">
+          <Event presentationTime="100" duration="0" id="0" messageData="foo"/>
+          <Event presentationTime="1100" duration="0" id="5" messageData="bar"/>
+          <Event presentationTime="2100" duration="0" id="6" messageData="foo_bar"/>
+        </EventStream>
+      </Period>
+    </MPD>`);
+  const expected = {
+    eventStream: [
+      {
+        end: 15,
+        id: '0',
+        messageData: 'foo',
+        schemeIdUri: 'urn:google:dai:2018',
+        start: 15,
+        value: undefined
```

Comment:
Is there a way to ensure we get the correct value here, or in one of the examples?

Reply:
Good call, I added a `value: "foo"` and modified one of the tests.